src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/visits/VisitService.kt

```diff
@@ -224,19 +224,25 @@ class VisitService(
     visit.visitors.removeAll(visitorsToRemove)
     visit.visitors.addAll(visitorsToAdd)
 
-    // copy the visitors to the visit order, ignoring the dummy visitor entry
-    visit.visitOrder?.let { visitOrder ->
-      visitOrder.visitors.clear()
-      visitOrder.visitors.addAll(
-        visit.visitors
-          .filter { it.person != null }.mapIndexed { index, visitor ->
-            VisitOrderVisitor(
-              person = visitor.person!!,
-              groupLeader = index == 0,
-              visitOrder = visitOrder,
-            )
-          },
-      )
+    visit.visitOrder?.also { visitOrder ->
```

Comment:
essentially the same type of code we do for visitors; just add/remove visitors to the order that only of they have actually been added or removed. Previous all rows were deleted and re-added